$FreeWRT$

	Authored by Thorsten Glaser <tg@mirbsd.de> under copyright
	assignment with the Free Software Foundation.

--- binutils-2.16.1/ld/lexsup.c.orig	2007-03-20 16:25:09.000000000 +0000
+++ binutils-2.16.1/ld/lexsup.c	2007-03-20 16:32:27.000000000 +0000
@@ -585,6 +585,14 @@ parse_args (unsigned argc, char **argv)
 
   ldemul_add_options (is, &shortopts, il, &longopts, irl, &really_longopts);
 
+  /* We support the -Wl,-foo idiom of the xgcc compiler driver here,
+     because some build system writers are too stupid to realise the
+     difference between LD and CCLD.  We do not support -Wl,-foo,bar
+     for simplicity; use -Wl,-foo -Wl,bar instead.  */
+  for (i = 1; i < argc; i++)
+    if (strncmp (argv[i], "-Wl,", 4) == 0)
+      memmove (argv[i], argv[i] + 4, strlen(argv[i] + 4) + /* NUL */ 1);
+
   /* The -G option is ambiguous on different platforms.  Sometimes it
      specifies the largest data size to put into the small data
      section.  Sometimes it is equivalent to --shared.  Unfortunately,
